include/boost/math/special_functions/detail/bessel_ik.hpp

```diff
@@ -374,7 +374,7 @@ int bessel_ik(T v, T x, T* I, T* K, int kind, const Policy& pol)
     current = Ku1;
     T scale = 1;
     T scale_sign = 1;
-    for (k = 1; k <= n; k++)                   // forward recurrence for K
+    for (k = 1; k <= n; k++)                   // forward recurrence for result_K
```

Comment:
Again K refers to the mathematical function not our variable... life would be easier if mathematicians didn't keep reusing the same names over and over!